src/module.c

```diff
@@ -11697,6 +11702,10 @@ int RM_RegisterEnumConfig(RedisModuleCtx *ctx, const char *name, int default_val
         enum_vals[i].name = zstrdup(enum_values[i]);
         enum_vals[i].val = int_values[i];
     }
+    /* For bitflags, sort them from high to low, so that if there are several / partially
+     * overlapping entries, we'll prefer the ones matching more bits. */
```

Comment:
If we sort, callers can't specify order preference for same value elements. An alternative is not to sort but require the input to be already sorted and in preference order.

Reply:
yes, i mentioned it... which one do you prefer?

Reply:
p.s. it's not that they "can't specify" the order... arguably there's no point in putting the lower ones first.
there are two scenarios where the order matters (taking the example from the test).
* one is that setting the config to `two four`, could result in a CONFIG GET that returns `four two` (i.e it controls the order, but not the composition)
* the other one is that `one four` could either result in `five`, `one four` or `four one`.

so regarding the actual composition (not the order of the reply), if the user defined a value of `five`, arguably, there's absolutely no reason to return `four` and `one`. so there's no reason to require the user to do the ordering on his side, we always prefer the higher values that match multiple bits.

Reply:
@oranagra I was thinking about `quattro` and `four` both assigned the same enum but the wants to specify the official is `four` and `quattro` is only an input alias. If we want to solve this by relying on order, we must not sort. We may still require non-identical enums to be sorted to address the composition issue you metnioned.

Reply:
I suppose we can make sure that our sorting code doesn't change the order of items with the same score. 
Or we can make the matching code to always prefer enums that much the most bits.

Reply:
for the record, i discussed this again with Yossi. we concluded we rather drop the sorting and document that the user defines the order.

p.s. another scenario that could be is an array of [one, two, three, four, five, six], a value of 7 could be either [one, six], or [two, five]. i.e. two possible values covering 2 bits.